src/main/java/org/jabref/gui/actions/CleanupAction.java

```diff
@@ -39,6 +43,26 @@ public CleanupAction(BasePanel panel, JabRefPreferences preferences) {
     }
 
     @Override
+    public void action() {
+        init();
+        Platform.runLater(() -> {
+            if (canceled) {
+                return;
+            }
+            CleanupDialog cleanupDialog = new CleanupDialog(panel.getBibDatabaseContext(), preferences.getCleanupPreset());
+
+            Optional<CleanupPreset> chosenPreset = cleanupDialog.showAndWait();
+            chosenPreset.ifPresent(cleanupPreset ->
+                    BackgroundTask
+                            .wrap(() -> {
+                                cleanup(cleanupPreset);
+                                return null;
+                            })
+                            .onSuccess(x -> SwingUtilities.invokeLater(this::showResults))
```

Comment:
the `showResults` method uses the JavaFX dialog service, so it should probably not run on the swing thread.